cpp/tests/io/parquet_test.cpp

```diff
@@ -1370,10 +1371,10 @@ TYPED_TEST(ParquetChunkedWriterNumericTypeTest, UnalignedSize2)
   auto filepath = temp_env->get_temp_filepath("ChunkedUnalignedSize2.parquet");
   cudf_io::chunked_parquet_writer_options args =
     cudf_io::chunked_parquet_writer_options::builder(cudf_io::sink_info{filepath});
-  auto state = cudf_io::write_parquet_chunked_begin(args);
-  cudf_io::write_parquet_chunked(tbl1, state);
-  cudf_io::write_parquet_chunked(tbl2, state);
-  cudf_io::write_parquet_chunked_end(state);
+  cudf_io::parquet_chunked_writer writer(args);
+  writer.write(tbl1);
+  writer.write(tbl2);
+  writer.write_end();
```

Comment:
I don't see a test where the destructor is used to flush the output instead of write_end. You can easily test this by placing the write section of the test in a black and omitting the `write_end` call.
I also don't see a negative test that throws if a write is called after `write_end`.